### ISSUE TITLE
Unity 2019.1.0a8 crash when using Compound Button Icon override icon function

### ISSUE THREAD

**Issue by dvdbot** [microsoft/MixedRealityToolkit-Unity#3213]
**Does this affect the legacy HoloToolkit (master) or the Mixed Reality Toolkit (mrtk_release)?**
HoloToolkit

**Describe the bug**
_A clear and concise description of what the bug is._
When you enable `Override Icon` and select a custom icon, it shows in the editor but when you press play...
the whole editor just crashes

**To Reproduce**
_(Links to sample GitHub project preferred)_
1. Create a new project
2. Add the default holographic button prefab or create one your own
3. Make sure override icon is enabled and an icon is selected
4. Press play to enjoy a editor crash

**Expected behavior**
_A clear and concise description of what you expected to happen._
Uhm... no crash?

**Actual behavior**
_A clear and concise description of what happens instead._
Crash

**Screenshots**
_If applicable, add screenshots to help explain your problem._
![image](https://user-images.githubusercontent.com/5251028/49446595-d7c5ec00-f7d4-11e8-9bb5-fbbc438e7e71.png)

**Unity Editor Version**
_Which Unity version(s) you've been able to reproduce the bug in._
Happens on all Unity Editor 2019.X version as far as I can see

**Mixed Reality Toolkit Release Version**
_Which MRTK version(s) you've been able to reproduce the bug in._
MRTKVersion.txt contains "2017.4.2.0"
**Additional context**
_Add any other context about the problem here._
It has to do with the following lines in `CompoundButtonIcon.cs`
```C#
167 // Use the default mesh for override icons
168 targetIconRenderer.enabled = true;
169 IconMeshFilter.sharedMesh = Profile.IconMesh;
170 IconMeshFilter.transform.localScale = Vector3.one;
171 instantiatedMaterial.mainTexture = iconOverride;
```
if I comment them out - no crash, but no icon... if I uncomment any of them, crash...

**Comment by Yoyozilla** [comment]
Bulk closing older issues. 
Please reopen if it still repros with the latest Unity 2018 + MRTK release

We fixed a few issues with MRTK and Unity 2019.1 lately. 
Also please note that the recommended Unity build for MRTK is Unity 2018.4 (LTS).

**Comment by Yoyozilla** [comment]
To add a little more background, we would like to recommend the LTS build of Unity for MRTK v2 given that servicing and no breaking change/feature change promise. 
We will continue to look into Unity 2019 dot releases and will eventually be releasing a verson of MRTK that will be compatible with Unity 2019 LTS when it comes out :)
